runtime/reconcilers/metrics_view.go

```diff
@@ -65,14 +65,22 @@ func (r *MetricsViewReconciler) Reconcile(ctx context.Context, n *runtimev1.Reso
 		return runtime.ReconcileResult{}
 	}
 
+	var modelConnector string
+	res, err := r.C.Get(ctx, &runtimev1.ResourceName{Name: mv.Spec.Table, Kind: runtime.ResourceKindModel}, false)
+	if err == nil && res.GetModel().State.ResultConnector != "" { // ignore error since dashboard can directly reference a table as well
+		modelConnector = res.GetModel().State.ResultConnector
+	} else {
+		modelConnector = mv.Spec.Connector
+	}
+
 	// NOTE: In other reconcilers, state like spec_hash and refreshed_on is used to avoid redundant reconciles.
 	// We don't do that here because none of the operations below are particularly expensive.
 	// So it doesn't really matter if they run a bit more often than necessary ¯\_(ツ)_/¯.
 
 	// NOTE: Not checking refs for errors since they may still be valid even if they have errors. Instead, we just validate the metrics view against the table name.
 
 	// Validate the metrics view and update ValidSpec
-	validateResult, validateErr := r.C.Runtime.ValidateMetricsView(ctx, r.C.InstanceID, mv.Spec)
+	validateResult, validateErr := r.C.Runtime.ValidateMetricsView(ctx, r.C.InstanceID, mv.Spec, modelConnector)
```

Comment:
There's already a `Connector` property on the spec, so better to update the spec before passing it to `ValidateMetricsView`. I.e. something like:
```go
// Enrich spec before validating it
spec := mv.Spec
if spec.Connector == "" {
  // Attempt to infer spec from a model
}

// ...
validateResult, validateErr := r.C.Runtime.ValidateMetricsView(ctx, r.C.InstanceID, spec)

Reply:
Isn't this equivalent to the solution that we had initially ? :)